### PR TITLE
Add a `--release` option

### PR DESCRIPTION
**Pull request by ghost**
`-p` means `--only-packages` plus a lot of other useful things to put `dune` in "reproducible release mode", such as:

- not reading the configuration file
- disabling root detection
- ...

Sometimes, we want the reproducible release mode without the package filtering. So this PR introduce a new option `--release` that means the same thing as `-p` except the package filtering. Or put it another way, `-p x` is now a shorthand for `--release --only-packages x`.

### CODE REVIEW

**Comment by emillon** [comment]
`--release` sounds a lot like `--profile release`, but it's also different so it might be a source of confusion. If that's properly documented, that might be ok though.

**Review by rgrinberg — APPROVED** [review]
Looks good to me. I agree that it's a bit confusing wrt `--profile release`. We should clarify that profiles are a separate thing from the true release mode.

**Comment by diml** [comment]
Yh, I was wondering about that as well. Would `--release-mode` be clearer?

**Comment by rgrinberg** [comment]
Tbh, `--release` still sounds better. I'm not sure `--release-mode` is any clearer to users.

**Comment by emillon** [comment]
Yes, I agree.

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+next (unreleased)
+-----------------
+
+- Add a `--release` option meaning the same as `-p` but without the
+  package filtering. This is useful for custom `dune` invocation in opam
+  files where we don't want `-p` (#...., @diml)
```
Comment (diml) [inline CHANGES.md]:
```suggestion
  files where we don't want `-p` (#3260, @diml)
```

**Comment by diml** [comment]
Alright, let's stick to `--release` then. Regarding the doc, maybe the part of the manual that mention `-p` could be improved. I'll have a look.

**Pull request merged**